## Problēmas apraksts

*Prognozēsim, kuras pasažieru grupas pēc vecuma un dzimuma visbiežāk tika prioritizētas evakuācijas laikā. Tas palīdzēs saprast, vai tika ievērots princips “sievietes un bērni vispirms”. Šī informācija ir svarīga kuģniecības kompānijai, lai izvērtētu ārkārtas rīcības efektivitāti un ētiskos aspektus. Prognozes būtu noderīgas drošības ekspertiem un apmācību veidotājiem.*


## ML uzdevuma tips

*Šis ir supervised uzdevums, jo mums ir zināms, kas izdzīvoja/neizdzīvoja. Un šī būs klasifikācija, jo modelis pasažieriem piešķirs vienu no divām vērtībām (0 Neizdzīvoja, 1 Izdzīvoja), bet, izmantojot vairāku kolonnu datus, piem., sex, age, sibsp, mācīsies atpazīt, kādiem pasažieriem bija lielāka izdzīvošanas varbūtība, un tāpēc varēspārbaudīt, vai princips "sievietes un bērni vispirms" tiešām tika ievērots.*


In [1]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

# Pārbaud
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [2]:
# Target mainīgais
print(df['Survived'].value_counts())
print(f'\nIzdzīvoja: {df["Survived"].mean():.1%}')

Survived
0    549
1    342
Name: count, dtype: int64

Izdzīvoja: 38.4%


## Target variable

*Target mainīgais ir tas, ko vēlamies prognozēt. Šajā gadījumā tas ir survived — kolonna, kas norāda, vai pasažieris izdzīvoja Titānika katastrofā.*

## Features saraksts

**Skaitliskās (numerical):**
age -> Vecums gados;
fare -> Biļetes cena;
sibsp -> Brāļu/māsu un laulāto skaits;
parch -> Vecāku/bērnu skaits.

**Kategoriskās (categorical):**
sex -> Dzimums;
pclass -> Biļetes klase (skaitliska);
embarked -> Iekāpšanas osta;
alone -> Vai ceļoja viens;
class -> Biļetes klase (vārdisks).

In [3]:
# Features pārskats
print('Skaitliskās features:')
print(df.select_dtypes(include='number').columns.tolist())
print('\nKategoriskās features:')
print(df.select_dtypes(include='str').columns.tolist())

Skaitliskās features:
['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']

Kategoriskās features:
['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


## Sākotnējie novērojumi

*Trūkstošās vērtības ir 3 kolonnās: 177 personām nezinām vecumu, 687 personām nezinām kajīšu numuriņus un 2 personām nav zināms, kurā ostā iekāpa. Cabin kolonnu sava uzdevuma ietvaros varu dzēst, jo nebūs izšķiroša vērtība, Embarked var aizpildīt ar populārāko vērtību, bet Age, kas man būtu ļoti svarīga kolonna, nāksies aizpildīt ar mediānu, nevis vidējo vērtību, lai nepievilktu ekstrēmās (galējās) vērtības.*

*Izdzīvoja 38.4% pasažieru.*

*Pagaidām neko pārsteidzošu vai interesantu nesaskatu.*

In [4]:
# Trūkstošās vērtības
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
